src/diffcalc_API/services/hkl.py

```diff
@@ -1,29 +1,44 @@
 from itertools import product
-from typing import Dict, List, Optional, Tuple, Union
+from typing import Dict, Iterator, List, Optional, Tuple, Union
 
 import numpy as np
 from diffcalc.hkl.geometry import Position
 
-from diffcalc_API.errors.hkl import InvalidMillerIndicesError, InvalidScanBoundsError
+from diffcalc_API.errors.hkl import (
+    InvalidAngleBoundsError,
+    InvalidMillerIndicesError,
+    InvalidScanBoundsError,
+)
 from diffcalc_API.models.ub import HklModel, PositionModel
 from diffcalc_API.stores.protocol import HklCalcStore
+from diffcalc_API.utils import async_wrap
 
 
 async def lab_position_from_miller_indices(
     name: str,
     miller_indices: HklModel,
     wavelength: float,
+    angles: Optional[List[str]],
+    low_bound: Optional[List[float]],
+    high_bound: Optional[List[float]],
```

Comment:
You pass these three parameters to several functions, it might be easier to tie them up in a single `dataclass` and pass that around e.g.

```python
@dataclass
class SolutionConstraints:
    axes: Optional[List[str]] = None
    low_bound: Optional[List[float]] = None
    high_bound: Optional[List[float]] = None
```

Reply:
Good idea, will implement